File: SkillFlow/SkillFlow/Models/Task.swift
//
//  Task.swift
//  SkillFlow
//
//  Created by SEEDO API Refactor on 2026/1/1.
//

import Foundation
import SwiftData

@Model
final class Task {
    @Attribute(.unique) var entryId: String
    var status: String // processing, audio_done, video_done, finished, failed
    var dirLocation: String
    var createdAt: Date
    var updatedAt: Date
    var errorMessage: String?
    
    init(entryId: String, dirLocation: String, status: String = "processing") {
        self.entryId = entryId
        self.dirLocation = dirLocation
        self.status = status
        self.createdAt = Date()
        self.updatedAt = Date()
    }
}

File: rust-backend/src/domain/task.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum TaskStatus {
    Processing,
    AudioDone,
    VideoDone,
    Finished,
    Failed,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Task {
    pub entry_id: String,
    pub dir_location: String,
    
    #[serde(skip_serializing_if = "Option::is_none")]
    pub transcript_text: Option<String>,
    
    #[serde(skip_serializing_if = "Option::is_none")]
    pub video_analysis: Option<serde_json::Value>,
    
    #[serde(skip_serializing_if = "Option::is_none")]
    pub steps_package: Option<serde_json::Value>,
    
    pub status: TaskStatus,
    
    #[serde(skip_serializing_if = "Option::is_none")]
    pub error: Option<String>,
    
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

impl Task {
    pub fn new(entry_id: String, dir_location: String) -> Self {
        let now = Utc::now();
        Self {
            entry_id,
            dir_location,
            transcript_text: None,
            video_analysis: None,
            steps_package: None,
            status: TaskStatus::Processing,
            error: None,
            created_at: now,
            updated_at: now,
        }
    }
}
